pkg/build/controller/image_change_controller.go

```diff
@@ -81,16 +80,23 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageRepository)
 
 		if shouldBuild {
 			glog.V(4).Infof("Running build for buildConfig %s in namespace %s", config.Name, config.Namespace)
-			b := buildutil.GenerateBuildFromConfig(config, nil, subs)
-			if err := c.BuildCreator.Create(config.Namespace, b); err != nil {
-				return fmt.Errorf("error starting build for buildConfig %s: %v", config.Name, err)
-			}
+			// first update the config
 			if err := c.BuildConfigUpdater.Update(config); err != nil {
```

Comment:
you should not be updating the BuildConfig until you've successfully created the build, otherwise you're setting the LastTriggeredImageID w/o necessarily actually having run that build and it'll never run again.